File: backend/src/main/java/cohappy/backend/model/dto/ModifyHouseDTO.java
package cohappy.backend.model.dto;

import lombok.Data;

import java.util.List;

@Data
public class ModifyHouseDTO {
    private String houseCode;
    private List<byte[]> images;
    private Integer costPerMonth;
    private String country;
    private String region;
    private String street;
    private Integer civicNumber;

    public boolean areAllNull(){
        return
                (images == null || images.isEmpty()) &&
                costPerMonth == null &&
                country == null &&
                region == null &&
                street == null &&
                civicNumber == null;
    }
}

File: frontend/app/src/main/java/cohappy/frontend/model/dto/request/ModifyHouseDTO.kt
package cohappy.frontend.model.dto.request
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class ModifyHouseDTO(
    val houseCode: String,
    val images: List<ByteArray>? = null,
    val costPerMonth: Int? = null,
    val country: String? = null,
    val region: String? = null,
    val street: String? = null,
    val civicNumber: Int? = null
) {
    fun areAllNull(): Boolean {
        return images.isNullOrEmpty() &&
                costPerMonth == null &&
                country == null &&
                region == null &&
                street == null &&
                civicNumber == null
    }
}